app/controllers/bookings_controller.rb

```diff
@@ -1,2 +1,44 @@
 class BookingsController < ApplicationController
+  before_action :set_offer, only: %i[new create]
+
+  def edit
+  end
+
+  def update
+    if @booking.update(params[:booking])
+      redirect_to bookings_path(@booking), notice: "Successfuly updated booking." #Again with the redirecting
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @booking.destroy
+  end
+
+  def new
+    @booking = Booking.new
+  end
+
+  def create
+    @booking = Booking.new(bookings_params)
+    @booking.offer = @offer
+    if @booking.save
```

Comment:
before the save, we should also indicate **@booking.user = current_user**, otherwise it won't be associated to a user and won't be saved.